### PR TITLE
🐛 Regenerate the hub kubeconfig secret if the cluster name of the current context changes

### PR DESCRIPTION
**Pull request by zhujian7**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/registration/spoke/spokeagent.go [hunk 1]
```diff
@@ -505,16 +507,20 @@ func (o *SpokeAgentConfig) HasValidHubClientConfig(ctx context.Context) (bool, e
...
 		return false, err
 	}
 
-	server, proxyURL, caBundle, err := parseKubeconfig(o.agentOptions.HubKubeconfigFile)
+	ctxCluster, server, proxyURL, caBundle, err := parseKubeconfig(o.agentOptions.HubKubeconfigFile)
 	switch {
 	case err != nil:
 		return false, err
-	case bootstrapServer != server, bootstrapProxyURL != proxyURL, !reflect.DeepEqual(bootstrapCABndle, caBundle):
+	case bootstrapCtxCluster != ctxCluster,
```
Comment (qiujian16) [inline pkg/registration/spoke/spokeagent.go:520]: we need some comment here or for the func on why we compare the clustername in the kubeconfig.

**Review by qiujian16 — COMMENTED** [review]
I think we need an integration or e2e to test the cluster name change

**Comment by qiujian16** [comment]
FYI @xuezhaojun this should also be considered in your PR

File: pkg/registration/spoke/spokeagent.go [hunk 2]
```diff
@@ -236,11 +236,12 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 			managementKubeClient, 10*time.Minute, informers.WithNamespace(o.agentOptions.ComponentNamespace))
 
 		// create a kubeconfig with references to the key/cert files in the same secret
-		srever, proxyURL, caData, err := parseKubeconfig(o.registrationOption.BootstrapKubeconfig)
+		contextClusterName, srever, proxyURL, caData, err := parseKubeconfig(o.registrationOption.BootstrapKubeconfig)
```
Comment (elgnay) [inline pkg/registration/spoke/spokeagent.go:239]: srever -> server

File: pkg/registration/spoke/spokeagent.go [hunk 3]
```diff
@@ -505,16 +507,26 @@ func (o *SpokeAgentConfig) HasValidHubClientConfig(ctx context.Context) (bool, e
 // 2. The proxy url
 // 3. The CA bundle
```
Comment (elgnay) [inline pkg/registration/spoke/spokeagent.go:508]: Add cluster name here as well

**Pull request merged**